systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaTopicUtils.java

```diff
@@ -281,8 +303,160 @@ public static void waitForTopicStatusMessage(String namespaceName, String topicN
         LOGGER.info("Waiting for KafkaTopic: {}/{} to contain message: {} in its status", namespaceName, topicName, message);
 
         TestUtils.waitFor(String.join("KafkaTopic: %s/%s status to contain message: %s", namespaceName, topicName, message), TestConstants.GLOBAL_POLL_INTERVAL, TestConstants.GLOBAL_TIMEOUT,
-            () -> KafkaTopicResource.kafkaTopicClient().inNamespace(namespaceName).withName(topicName).get()
-                .getStatus().getConditions().stream().anyMatch(condition -> condition.getMessage().contains(message))
+                () -> KafkaTopicResource.kafkaTopicClient().inNamespace(namespaceName).withName(topicName).get()
+                        .getStatus().getConditions().stream().anyMatch(condition -> condition.getMessage().contains(message))
         );
     }
+
+    /**
+     * Waits for a specific condition to be met on a KafkaTopic within a namespace.
+     *
+     * @param waitDescription   A human-readable description of the condition being waited on.
+     * @param namespaceName     The Kubernetes namespace in which the KafkaTopic resides.
+     * @param topicName         The name of the KafkaTopic to check.
+     * @param condition         A Predicate that takes a KafkaTopic and returns true if the condition is met.
+     */
+    public static void waitForCondition(String waitDescription, String namespaceName, String topicName, Predicate<KafkaTopic> condition) {
```

Comment:
I assume that this can be actually private you are using this exclusively within the Class.